ratatui/src/layout/rect.rs

```diff
@@ -166,6 +187,29 @@ impl Rect {
         }
     }
 
+    /// Resizes the `Rect` without modifying its position.
+    ///
+    /// Resizes the `Rect` by a given delta without modifying its [`x`](Rect::x)
+    /// or [`y`](Rect::y) coordinates.
+    /// - Positive `width` increases the extent of the `Rect` along the x-axis, negative `width`
+    ///   decreases it.
+    /// - Positive `height` increases the extent of the `Rect` along the y-axis, negative `height`
+    ///   decreases it.
+    ///
+    /// See [`SizeDelta`] for details.
+    #[must_use = "method returns the modified value"]
+    pub fn resize_by(self, delta: SizeDelta) -> Self {
```

Comment:
Can you please add a note about the expected behavior when clamping?
Something like "The size is clamped to ensure that the right and bottom bounds of the rect are less than u16::Max" (Feel free to wordsmith that if you want)

Reply:
I added a note